src/Template/Layout/js/app/components/folder-picker/folder-picker.js

```diff
@@ -0,0 +1,83 @@
+import { Treeselect, LOAD_ROOT_OPTIONS, LOAD_CHILDREN_OPTIONS } from '@riophae/vue-treeselect'
+import '@riophae/vue-treeselect/dist/vue-treeselect.css'
+import { t } from 'ttag';
+import { EventBus } from '../../directives/eventbus.js';
+
+const API_URL = new URL(BEDITA.base).pathname;
+const API_OPTIONS = {
+    credentials: 'same-origin',
+    headers: {
+        'accept': 'application/json',
+    }
+};
+
+export default {
+
+    components: {
+        Treeselect,
+    },
+
+    template: `<div>
+        <Treeselect placeholder="${t`Folder`}" :options="options" :load-options="loadOptions" v-model="value" :disabled="disabled" />
+        <input type="hidden" name="folderSelected" :value="value" />
+    </div>`,
+
+    data() {
+        return {
+            options: null,
+            value: null,
+            disabled: true,
+        };
+    },
+
+    mounted() {
+        EventBus.$on('folder-picker-init', this.initOptions);
+    },
+
+    beforeDestroy() {
+        EventBus.$off('folder-picker-init', this.initOptions);
+    },
+
+    methods: {
+
+        async fetchFolders(parent) {
+            const folders = [];
+            let filter = !parent ? 'filter[roots]' : `filter[parent]=${parent.id}`;
+            let page = 1;
+            let i = 0;
+            do {
```

Comment:
maybe calls after the first one can be parallelized (without the `await`)? the pagination infos can be fetched with the first call, then the while doesn't need the `true` condition but can be something like
```js
while(page <= paging.page_count)
```

Reply:
it's a remote case. "limit 100": more than 100 children (on first level). pagination is there, just in case...

Reply:
a second thought: better a refactor to use the pattern suggested. done ;)